src/functionalTest/groovy/com/autonomousapps/android/AbstractAndroidSpec.groovy

```diff
@@ -41,6 +42,7 @@ abstract class AbstractAndroidSpec extends AbstractFunctionalSpec {
     AGP_8_4,
     AGP_8_5,
     AGP_8_6,
+    AGP_8_7,
```

Comment:
The problem is this is too many versions. CI can't handle this load. We should test against the min version (8.0), the current stable version (8.5.1), and maybe also the latest alpha.

Reply:
Thanks for the tip!
I've updated the PR to test only against AGP 8.0 (min supported stable version), 8.5 (max supported stable version), 8.7 (latest alpha version).

Reply:
It's still failing. I think it was fine running the tests for every AGP versions, but there is probably some breaking change in AGP 8.7

Reply:
It looks like AGP 8.7 requires Gradle 8.9. So I've updated the project to use Gradle 8.9 (previously 8.7) and made the CI run tests for every AGP version again.
Let me know if you still prefer to keep only the min stable version, max stable version and the latest alpha version.

Reply:
It seems that everything is working fine now.
- Is it ok to upgrade to Gradle 8.9, or do you prefer a separate PR?
- Should I keep the tested AGP versions list as it is now? Or only the three versions you suggested?

Reply:
I think updating to Gradle 8.9 would be better in a separate PR.

For this PR, you could just update the AGP/Gradle compatibility matrix, and then it won't try to run the test against an incompatible combination.

Reply:
I've created the PR to update Gradle here: https://github.com/autonomousapps/dependency-analysis-gradle-plugin/pull/1236

I'll update this PR soon too.

Reply:
It should be good now. Let me know if I missed something.